lib/jormungandr/src/Cardano/Wallet/Jormungandr/Network.hs

```diff
@@ -147,30 +150,43 @@ import qualified Data.Text.Encoding as T
 -- | Creates a new 'NetworkLayer' connecting to an underlying 'Jormungandr'
 -- backend target.
 newNetworkLayer
-    :: forall n. ()
-    => BaseUrl
-    -> IO (NetworkLayer (Jormungandr n) IO)
+    :: BaseUrl
+    -> IO (NetworkLayer IO Tx (Block Tx))
 newNetworkLayer = fmap snd . newNetworkLayer'
 
 -- | Creates a new 'NetworkLayer' connecting to an underlying 'Jormungandr'
 -- backend target. Also returns the internal 'JormungandrLayer' component.
 newNetworkLayer'
-    :: forall n. ()
-    => BaseUrl
-    -> IO (JormungandrLayer IO, NetworkLayer (Jormungandr n) IO)
+    :: BaseUrl
+    -> IO (JormungandrLayer IO, NetworkLayer IO Tx (Block Tx))
 newNetworkLayer' url = do
     mgr <- newManager defaultManagerSettings
     st <- newMVar emptyUnstableBlocks
     let jor = mkJormungandrLayer mgr url
     return (jor, mkNetworkLayer st jor)
 
+
 -- | Wrap a Jormungandr client into a 'NetworkLayer' common interface.
 mkNetworkLayer
     :: MonadBaseControl IO m
     => MVar UnstableBlocks
     -> JormungandrLayer m
-    -> NetworkLayer (Jormungandr n) m
-mkNetworkLayer st j = NetworkLayer
+    -> NetworkLayer m Tx (Block Tx)
+mkNetworkLayer st j = coerceBlock <$> mkRawNetworkLayer st j
+
+-- | Wrap a Jormungandr client into a 'NetworkLayer' common interface.
+--
+-- This version provides the full, raw blocks from
+-- "Cardano.Wallet.Jormungandr.Binary".
+--
+-- TODO: We may want to aim to replace the other constructors and only keep
+-- this one.
+mkRawNetworkLayer
+    :: MonadBaseControl IO m
+    => MVar UnstableBlocks
+    -> JormungandrLayer m
+    -> NetworkLayer m Tx J.Block
+mkRawNetworkLayer st j = NetworkLayer
```

Comment:
I wouldn't mind spending some time refactoring a bit the constructors for Jormungandr network layer and the way things are articulated. At the moment it's a bit of a mess and it's hard to figure out which one is used for what. 

I guess that's a necessary evil for now, but once we have a clearer idea of what are the needs regarding the network layer, I'd definitely like us to review the top-level module's API to something more sensible.